Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.inl

```diff
@@ -91,22 +91,21 @@ namespace AZ::Dom
     }
 
     template<class T>
-    bool DomPrefixTree<T>::AttachNodeAtPath(const Path& path, Node&& nodeToAttach)
+    bool DomPrefixTree<T>::OverwritePath(const Path& path, Node&& newNode, bool createParentsIfNeeded)
     {
         Node* node = &m_rootNode;
-        const size_t entriesToIterate = path.Size() - 1;
-        for (size_t i = 0; i < entriesToIterate; ++i)
+        const size_t parentEntriesToIterate = path.Size() - 1;
+        for (size_t i = 0; i < parentEntriesToIterate; ++i)
         {
             const PathEntry& entry = path[i];
-            auto nodeIt = node->m_values.find(entry);
-            if (nodeIt == node->m_values.end())
+            if (!createParentsIfNeeded && node->m_values.find(entry) == node->m_values.end())
             {
                 return false;
             }
-            node = &nodeIt->second;
+            node = &node->m_values[entry]; // get or create an entry in this node
         }
 
-        node->m_values[path[path.Size() - 1]] = AZStd::move(nodeToAttach);
+        node->m_values[path[path.Size() - 1]] = AZStd::move(newNode);
```

Comment:
This can also create a new node when 'createParentsIfNeeded' is false. That can be confusing imo. We can't overwrite something that didn't exist in the first place. While we are here cleaning up this logic, might as well cover all bases.

I recommend returning false here if node->m_values[path[path.Size() - 1]]  is empty.